authzed/api/v1alpha1/lookupwatch_service.proto

```diff
@@ -0,0 +1,78 @@
+syntax = "proto3";
+package authzed.api.v1alpha1;
+
+option go_package = "github.com/authzed/authzed-go/proto/authzed/api/v1alpha1";
+option java_package = "com.authzed.api.v1alpha1";
+
+import "google/api/annotations.proto";
+import "validate/validate.proto";
+
+import "authzed/api/v1/core.proto";
+
+// LookupWatchService is used to receive a stream of updates for resources of a
+// specific (resource type, permission, subject) combination.
+service LookupWatchService {
+
+  // WatchAccessibleResources initiates a watch for accessibility changes for the provided
+  // (resource type, permission, subject) pair.
+  rpc WatchAccessibleResources(WatchAccessibleResourcesRequest)
+    returns (stream WatchAccessibleResourcesResponse) {
+      option (google.api.http) = {
+        post: "/v1alpha1/lookupwatch"
+        body: "*"
+      }; 
+    }
+}
+
+// WatchAccessibleResourcesRequest starts a watch for specific permission updates 
+// for the given resource and subject types.
+message WatchAccessibleResourcesRequest {
+
+  // resource_object_type is the type of resource object for which we will
+  // watch for changes.
+  string resource_object_type = 1 [ (validate.rules).string = {
+    pattern : "^([a-z][a-z0-9_]{2,61}[a-z0-9]/)?[a-z][a-z0-9_]{2,62}[a-z0-9]$",
+    max_bytes : 128,
+  } ];
+
+  // permission is the name of the permission or relation for which we will
+  // watch for changes.
+  string permission = 2 [ (validate.rules).string = {
+    pattern : "^[a-z][a-z0-9_]{2,62}[a-z0-9]$",
+    max_bytes : 64,
+  } ];
+
+  // subject_object_type is the type of the subject resource for which we will
+  // watch for changes.
+  string subject_object_type = 3;
+
+  // optional_subject_relation allows you to specify a group of subjects to watch
+  // for a given subject type.
+  string optional_subject_relation = 4;
+
+  authzed.api.v1.ZedToken optional_start_cursor = 5;
+}
+
+// PermissionUpdate represents a single permission update for a specific
+// subject's permissions.
+message PermissionUpdate {
+  
+  // todo: work this into the v1 core API at some point since it's used
+  // accross services.
+  enum Permissionship {
+    PERMISSIONSHIP_UNSPECIFIED = 0;
+    PERMISSIONSHIP_NO_PERMISSION = 1;
+    PERMISSIONSHIP_HAS_PERMISSION = 2;
+  }
+
+  // subject defines the subject resource whose permissions have changed.
+  authzed.api.v1.SubjectReference subject = 1;
+  authzed.api.v1.ObjectReference resource = 2;
```

Comment:
docstrings for every field help a ton since they are surfaced in the buf registry documentation

Reply:
What is the full expectation in this regard? I see a lot of scenarios where things like `string relation = X;` aren't commented or top level proto messages etc..

Are you just asking for a docstring on this particular line for the `resource` definition, or do you want a docstring on every line?

Reply:
Ideally every field in a message is documented with a string, but I understand that this is 1) still experimental and 2) a higher standard than existing APIs.

Because the buf registry surfaces docstrings for each message, I think at a minimum we should have those.

Reply:
Sounds good, thanks for the clarification.

I'd updated the proto definition to include the docstring